.github/workflows/ci.yml

```diff
@@ -18,97 +18,83 @@ jobs:
             java: 8
           - os: macOS-latest
             java: 8
+        include:
+          - os: ubuntu-latest
+            CACHE_PATH: ~/.cache/coursier/v1
+          - os: macOS-latest
+            CACHE_PATH: ~/Library/Caches/Coursier/v1
+          - os: windows-latest
+            CACHE_PATH: ~\AppData\Local\Coursier\Cache\v1
     steps:
       - uses: actions/checkout@v1
       - uses: olafurpg/setup-scala@v7
         with:
           java-version: "adopt@1.${{ matrix.java }}"
+      - uses: actions/cache@v1
+        with:
+          path: $CACHE_PATH
+          key: ${{ runner.os }}-coursier-${{ hashFiles('**/*.sbt') }}
+          restore-keys: |
+            ${{ runner.os }}-coursier-
+      - uses: actions/cache@v1
+        with:
+          path: ~/.sbt
+          key: ${{ runner.os }}-sbt-${{ hashFiles('project/build.properties') }}-${{ hashFiles('project/plugins.sbt') }}
+          restore-keys: ${{ runner.os }}-sbt-
       - name: Run unit tests
         run: |
           bin/test.sh unit/test
         shell: bash
-  sbt:
-    name: Sbt integration
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v1
-      - uses: olafurpg/setup-scala@v7
-        with:
-          java-version: "adopt@1.11"
-      - name: Run Sbt tests
-        run: bin/test.sh 'slow/testOnly -- tests.sbt'
-  maven:
-    name: Maven integration
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v1
-      - uses: olafurpg/setup-scala@v7
-        with:
-          java-version: "adopt@1.11"
-      - name: Run Maven tests
-        run: bin/test.sh 'slow/testOnly -- tests.maven'
-  gradle:
-    name: Gradle integration
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v1
-      - uses: olafurpg/setup-scala@v7
-        with:
-          java-version: "adopt@1.11"
-      - name: Run Gradle tests
-        run: bin/test.sh 'slow/testOnly -- tests.gradle'
-  mill:
-    name: Mill integration
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v1
-      - uses: olafurpg/setup-scala@v7
-        with:
-          java-version: "adopt@1.11"
-      - name: Run Mill tests
-        run: bin/test.sh 'slow/testOnly -- tests.mill'
-  pants:
-    name: Pants integration
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v1
-      - uses: olafurpg/setup-scala@v7
-        with:
-          java-version: "adopt@1.11"
-      - name: Run Pants tests
-        run: bin/test.sh 'slow/testOnly -- tests.pants'
-  feature:
-    name: LSP integration tests
+  integration:
+    name: ${{ matrix.name }}
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        type:
+          [sbt, maven, gradle, pants, mill, feature, cross, scalafmt, scalafix]
+        include:
+          - type: sbt
```

Comment:
It should be easy now to split the unit tests, but I don't want to do too much in this PR. Will follow up later